pkgs/development/go-packages/generic/default.nix

```diff
@@ -130,7 +137,7 @@ let
         echo "$d" | grep -q "\(/_\|examples\|Godeps\)" && return 0
         [ -n "$excludedPackages" ] && echo "$d" | grep -q "$excludedPackages" && return 0
         local OUT
-        if ! OUT="$(go $cmd $buildFlags "''${buildFlagsArray[@]}" -v -p $NIX_BUILD_CORES $d 2>&1)"; then
+        if ! OUT="$(CC=$CC_FOR_TARGET CXX=$CXX_FOR_TARGET go $cmd $buildFlags "''${buildFlagsArray[@]}" -v -p $NIX_BUILD_CORES $d 2>&1)"; then
```

Comment:
if `stdenv.buildPlatform != stdenv.targetPlatform` then `CC_FOR_TARGET` becomes `null`. This would lead to CC not being set at all. Should `$CC` not have the correct value anyway?

Reply:
True. I will add checking for this corner case.

Reply:
Also check how this is different from the current value `$CC`.

Reply:
@Mic92 I notice the following. Take `pkgsCross.aarch64-multiplatform.ws` as an example. If you inspect `pkgsCross.aarch64-multiplatform.ws.stdenv.hostPlatform.config`, you get `x86_64-unknown-linux-gnu` on a Linux machine. This platform triple looks wrong. The reason could be cause `buildGoPackage` actually uses `go.stdenv.mkDerivation` instead of just `stdenv`.